### PR TITLE
Modification of Karpenter Grafana dashboards capacity and performance

### PR DESCRIPTION
**Pull request by youwalther65**
Fixes [Karpenter 0.35.2 Grafana dashboards do not work properly #5934](https://github.com/aws/karpenter-provider-aws/issues/5934)

**Description**

- Capacity dashboard: removed deprecated metric "karpenter_deprovisioning_actions_performed" and provides new visualizations using metrics "karpenter_disruption_nodes_disrupted_total" and "karpenter_disruption_eligible_nodes"
- Performance dashboard: filtered for job="karpenter" to avoid getting controllers from other scarep jobsof K8s components using controller-runtime metrics as well

**How was this change tested?**

- Karpenter serviceMonitor object for kube-prometheus-stack created, which creates a scarep job "karpenter"
- provisioned dashboards to local Grafana instance

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by njtran — COMMENTED** [review]
I haven't pulled this down and applied it to my cluster, but I'd expect to see instances of `karpenter_deprovisioning_actions_performed` replaced with `karpenter_disruption_actions_performed`. But I don't see that. Can we just do a find all replace with the existing deprovisioning metrics and add the new metrics/changes in another PR?

**Comment by youwalther65** [comment]
@njtran I used metric [karpenter_disruption_nodes_disrupted_total](https://karpenter.sh/docs/reference/metrics/#karpenter_disruption_nodes_disrupted_total) instead of metric [karpenter_disruption_actions_performed_total](https://karpenter.sh/docs/reference/metrics/#karpenter_disruption_actions_performed_total).
Both provide drill-down by label disruption action, method, and consolidation type, but [karpenter_disruption_nodes_disrupted_total](https://karpenter.sh/docs/reference/metrics/#karpenter_disruption_nodes_disrupted_total) gives these details even by NodePool, which I believe is a clear advantage over [karpenter_disruption_actions_performed_total](https://karpenter.sh/docs/reference/metrics/#karpenter_disruption_actions_performed_total)

**Comment by njtran** [comment]
Hey @youwalther65, the number of actions taken is different from the number of nodes disrupted. Actions taken can provide information on the number of batches of disruptions. Do you mind adding the number of disruption actions back in as well?

**Comment by youwalther65** [comment]
Hey @njtran , that's interesting, thank you. So it would look something like that:
Screenshot 2024-04-09 at 12 19 25
with the other Visualizations from the originla still in place (and not visible in the screenshot I provided). If you like that I will go ahead and put it into my PR.

**Comment by jmdeal** [comment]
While disruption can be controlled at the nodepool level (TTLs, budgets, etc) the actual disruption actions are NodePool agnostic. For example, a single multi-node consolidation action could act upon nodes owned by multiple NodePools. We don't consider any action tied to a particular nodepool for that reason.

**Comment by jmdeal** [comment]
Sorry it took me a moment to get around to pulling this down and reviewed, overall really like the changes!

Some nits on the chart naming for consistency / spelling:
- Nodes eligable for Disruptions -> Nodes Eligible for Disruption
- Number of disruption actions performed -> Disruption Actions Performed
- Node Disruptions -> Voluntary Node Disruptions

I was going to suggest "Nodes Disrupted" for the last one to be consistent with the other naming, but I realized that's really ambiguous since we also have "Nodes Terminated". Since these dashboards are really intended for newcomers to Karpenter, I don't think we want to rely on terminology that requires knowledge of Karpenter's node lifecycle to differentiate between. I think voluntary node disruptions hits the mark since it filters out involuntary disruptions which aren't driven by the disruption controller (e.g. spot interruption, kubectl delete node, etc). Definitely open to other ideas here though.

Some other nitty feedback:
- IMO nodes eligible for disruption shouldn't be stacked since the same node can be counted in multiple categories. Maybe I'm wrong but I've always read stacked graphs as the different sections being mutually exclusive.
- Now that we have multiple charts regarding disruption here, I'm wondering if it would be good to add a dashboard filter for the disruption method? Seems like it would be a nice addition if you're interested in adding it.

I did also notice some funky numbers for nodes eligible for disruption, turns out there's a bug in this metric that should be fixed with the following PR: https://github.com/kubernetes-sigs/karpenter/pull/1187.

**Comment by youwalther65** [comment]
@jmdeal Thank you for the feedback. I like the ideas and have incorporated all of them in a new version. Please check again, thx.

**Comment by jmdeal** [comment]
/unstale I'll try and take a look by the end of the day or tomorrow, I got caught up with some other tasks last week and this slipped through the cracks. Need to get the upstream metric fix in as well.

**Comment by jmdeal** [comment]
Sorry for the delays here, two last comments and this looks good to me:
- Looks like there's still a typo in "Nodes Eligibale for Disruption" (should be Eligible)
- We should probably copy these dashboards into the v0.35, v0.36, and preview doc folders as well. Shouldn't need any additional changes there, just copies.

**Comment by youwalther65** [comment]
@jmdeal Thx, will follow up after my vacation end of  May.

**Comment by youwalther65** [comment]
Done with the requested changes @jmdeal

File: website/content/en/docs/getting-started/getting-started-with-karpenter/karpenter-capacity-dashboard.json [hunk 1]
```diff
@@ -1,1440 +1,1743 @@
...
+				"AliasAttributes": [
+					"email",
+					"preferred_username"
+				],
+				"UserPoolName": {
+					"Fn::Sub": "${AWS::StackName}-UserPool"
+				}
+			}
+		}
+	}
```
Comment (jmdeal) [inline website/content/en/docs/getting-started/getting-started-with-karpenter/karpenter-capacity-dashboard.json:1742]: I'm curious where this came from, seems like it was added in this latest commit. I can't figure out what, if anything, this affects in the dashboard but it seems out of place and looks more like cloudformation json than anything.

Comment (youwalther65) [reply website/content/en/docs/getting-started/getting-started-with-karpenter/karpenter-capacity-dashboard.json]: That's really kind of strange. I am not aware how this happened, sorry for that. Removed it , should work now.

**Review by jmdeal — APPROVED** [review]
LGTM, thanks for the contribution 🚀

**Pull request merged**